### PR TITLE
Fix type-only import declaration with empty named imports being parsed as having a default import specifier

### PR DESCRIPTION
**Pull request by dsherret**
As title.

Sorry, I missed this until now.

### CODE REVIEW

File: ecmascript/parser/src/parser/stmt/module_item.rs [hunk 1]
```diff
@@ -60,8 +60,13 @@ impl<'a, I: Tokens> Parser<'a, I> {
             .map(ModuleItem::from);
         }
 
-        let type_only_span = self.input.cur_span();
-        let type_only = self.syntax().typescript() && eat!("type");
+        let type_only = self.input.syntax().typescript()
+            && is!("type")
+            && (peeked_is!('{') || !peeked_is!("from") && !peeked_is!(','));
```
Comment (dsherret) [inline ecmascript/parser/src/parser/stmt/module_item.rs:65]: Wait... `import type from from "./a"` 🤯

That's actually a bug in the typescript compiler.

Comment (dsherret) [reply ecmascript/parser/src/parser/stmt/module_item.rs:65]: I'm not sure about a way to do a peek twice or what would be best to fix this. Overall it's not too important and such an edge case... going to sleep now.

Comment (kdy1) [reply ecmascript/parser/src/parser/stmt/module_item.rs:65]: I think the best solution is just ignoring it.
I don't expect a user to write code like `import type from 'foo';`

**Review by kdy1 — APPROVED** [review]
Great! Thanks!

**Pull request merged**